tests/swap_happy_path.rs

```diff
@@ -1,313 +1,106 @@
-use elements_rpc::{Client, ElementsRpc};
-use secp256k1::rand::thread_rng;
-use secp256k1::{Message, SecretKey, SECP256K1};
-use testcontainers::clients::Cli;
+extern crate link_cplusplus;
 
-#[tokio::test]
-async fn collaborative_create_and_sign() {
-    let tc_client = Cli::default();
-    let (client, _container) = {
-        let blockchain = Elementsd::new(&tc_client, "0.18.1.9").unwrap();
-
-        (
-            Client::new(blockchain.node_url.clone().into()).unwrap(),
-            blockchain,
-        )
-    };
+use baru::swap::{alice_finalize_transaction, bob_create_transaction, Actor};
+use elements::bitcoin::Amount;
+use elements::{bitcoin, Address, AddressParams};
+use rand::{CryptoRng, RngCore, SeedableRng};
+use rand_chacha::ChaChaRng;
+use secp256k1::SECP256K1;
+use tokio::sync::Mutex;
+use util::{make_keypair, verify_wallet_inputs_spend, Wallet};
 
-    let asset_id_lbtc = client.get_bitcoin_asset_id().await.unwrap();
-    let asset_id_usdt = client.issueasset(10.0, 0.0, true).await.unwrap().asset;
+mod util;
 
-    // fund keypairs and addresses
-    let (
-        fund_address_alice,
-        fund_sk_alice,
-        _fund_pk_alice,
-        fund_blinding_sk_alice,
-        _fund_blinding_pk_alice,
-    ) = make_confidential_address();
-    let (fund_address_bob, fund_sk_bob, _fund_pk_bob, fund_blinding_sk_bob, _fund_blinding_pk_bob) =
-        make_confidential_address();
-
-    // redeem keypairs and addresses
-    let (
-        final_address_alice,
-        _final_sk_alice,
-        _final_pk_alice,
-        _final_blinding_sk_alice,
-        _final_blinding_pk_alice,
-    ) = make_confidential_address();
-    let (
-        final_address_bob,
-        final_sk_bob,
-        _final_pk_bob,
-        final_blinding_sk_bob,
-        _final_blinding_pk_bob,
-    ) = make_confidential_address();
-
-    // initial funding
-    let fund_amount_alice = bitcoin::Amount::ONE_BTC;
-    let fund_amount_bob = bitcoin::Amount::ONE_BTC;
+#[tokio::test]
+async fn collaborative_create_and_sign() {
+    let mut rng = ChaChaRng::seed_from_u64(0);
 
-    let fund_alice_txid = client
-        .send_asset_to_address(&fund_address_alice, fund_amount_alice, Some(asset_id_lbtc))
-        .await
+    let btc_asset_id = "0c5d451941f37b801d04c46920f2bc5bbd3986e5f56cb56c6b17bedc655e9fc6"
+        .parse()
         .unwrap();
-
-    let fund_bob_txid = client
-        .send_asset_to_address(&fund_address_bob, fund_amount_bob, Some(asset_id_usdt))
-        .await
+    let usdt_asset_id = "6b397062b69411b554ec398ae3b25fdc54fab1805126786581a56a7746afbab2"
+        .parse()
         .unwrap();
 
     let amount_alice = bitcoin::Amount::from_sat(50_000_000);
     let amount_bob = bitcoin::Amount::from_sat(25_000_000);
 
-    let input_alice = extract_input(
-        &client.get_raw_transaction(fund_alice_txid).await.unwrap(),
-        fund_address_alice,
-    )
-    .unwrap();
+    let alice_wallet = Mutex::new(Wallet::default());
```

Comment:
Why is this wallet wrapped in a `Mutex`? I don't see any concurrency in this test so you should be fine with a `&mut`?